Mailer/app/api/v1/Handlers/Users/IsUserUnsubscribedHandler.php

```diff
@@ -2,40 +2,35 @@
 
 namespace Remp\MailerModule\Api\v1\Handlers\Users;
 
-use Nette\Utils\Json;
-use Nette\Utils\JsonException;
-use Remp\MailerModule\Repository\ListsRepository;
-use Remp\MailerModule\Repository\LogsRepository;
-use Remp\MailerModule\Repository\MailTypesRepository;
 use Remp\MailerModule\Repository\UserSubscriptionsRepository;
 use Tomaj\NetteApi\Handlers\BaseHandler;
-use Tomaj\NetteApi\Params\InputParam;
 use Tomaj\NetteApi\Response\JsonApiResponse;
-use Tracy\Debugger;
+use Remp\MailerModule\Api\Validation\InputValidatorTrait;
 
 class IsUserUnsubscribedHandler extends BaseHandler
 {
     private $userSubscriptionsRepository;
 
+    use InputValidatorTrait;
+
     public function __construct(
         UserSubscriptionsRepository $userSubscriptionsRepository
     ) {
         parent::__construct();
         $this->userSubscriptionsRepository = $userSubscriptionsRepository;
     }
 
-    public function params()
+    public function handle($payload)
     {
-        return [
-            new InputParam(InputParam::TYPE_POST, 'user_id', InputParam::REQUIRED),
-            new InputParam(InputParam::TYPE_POST, 'mail_type_id', InputParam::REQUIRED),
-        ];
-    }
+        $request = file_get_contents("php://input");
+        $payload = $this->validateInput($request, 'is-user-unsubscribed.schema.json');
 
-    public function handle($params)
-    {
-        $output = $this->userSubscriptionsRepository->isUserUnsubscribed($params['user_id'], $params['mail_type_id']);
+        if ($this->hasErrors()){
+            return $payload;
+        }
+
+        $output = $this->userSubscriptionsRepository->isUserUnsubscribed($payload['user_id'], $payload['mail_type_id']);
 
-        return new JsonApiResponse(200, ['status' => 'ok', 'data' => $output]);
+        return new JsonApiResponse(200, ['data' => $output]);
```

Comment:
By removing `status => ok` I meant not to wrap responses into arbitrary `data` object. API client knows it;s *data*, no need to tell that by putting results into `data` property.

Reply:
Yes, I removed it on other places, I must have missed this one.